OpenRA.Mods.Common/Traits/SupportPowers/ParatroopersPower.cs

```diff
@@ -227,5 +219,27 @@ public Actor[] SendParatroopers(Actor self, WPos target, bool randomize = true,
 
 			return units.ToArray();
 		}
+
+		private void RemoveCamera(Actor camera)
+		{
+			if (camera == null)
+				return;
+
+			camera.QueueActivity(new Wait(info.CameraRemoveDelay));
+			camera.QueueActivity(new RemoveSelf());
+			camera = null;
+		}
+
+		private void RemoveBeacon(Beacon beacon)
+		{
+			if (beacon == null)
+				return;
+
+			Self.World.AddFrameEndTask(w =>
+			{
+				w.Remove(beacon);
```

Comment:
I wonder if this could be an issue when the beacon is removed between the `beacon == null` check and this frame end task running.

Reply:
This is how it is now. I just copied the code. Should I move the `null` check inside the frame end task?

Reply:
Add another one. Not queuing a frame end task when `beacon` is `null` still makes sense, so that check doesn't have to go, imho.

Reply:
Amended.